libs/estdlib/src/lists.erl

```diff
@@ -563,9 +573,9 @@ duplicate(Count, Elem, Acc) -> duplicate(Count - 1, Elem, [Elem | Acc]).
 %%-----------------------------------------------------------------------------
 -spec sublist([Elem], integer()) -> [Elem].
 sublist(List, Len) when is_integer(Len) andalso Len >= 0 ->
-    sublist0(List, Len, []).
+    sublist0(List, Len).
 
 %% @private
-sublist0(_List, 0, Acc) -> reverse(Acc);
-sublist0([], _, Acc) -> reverse(Acc);
-sublist0([H | Tail], Len, Acc) -> sublist0(Tail, Len - 1, [H | Acc]).
+sublist0([], _Len) -> [];
+sublist0(_, 0) -> [];
+sublist0([H | Tail], Len) -> [H | sublist0(Tail, Len - 1)].
```

Comment:
Also here — are we losing tail recursion optimization?

Reply:
Old version was calling reverse which is not tail recursive.
New version is not tail recursive without calling reverse.